core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/CsdLocationFilter.java

```diff
@@ -72,17 +74,26 @@ private boolean allRequiredCsdUrlLocationEligible(Map<String, Set<String>> csdLi
         List<String> imagePreWarmCsd = image.getPreWarmCsd();
         LOGGER.debug("Available CSD urls: {}", imagePreWarmCsd);
         return requiredParcels.stream()
+                .filter(parcel -> isCsdExistsOnImage(imagePreWarmCsd, parcel))
                 .collect(Collectors.toMap(
                         parcelName -> parcelName,
                         parcelName -> filterCsdUrlsByStackRelatedParcel(imagePreWarmCsd, parcelName)));
     }
 
+    private boolean isCsdExistsOnImage(List<String> imagePreWarmCsd, String parcelName) {
```

Comment:
minor: naming is a little awkward, `is` is unnecessary.

Reply:
Idea insisted that a boolean should start with is/doest etc..